Sources/Player/PlaybackEngine/PlayerEngine.swift

```diff
@@ -228,6 +228,10 @@ final class PlayerEngine {
 				return
 			}
 
+			guard self.nextItem?.mediaProduct != mediaProduct else {
+				return
+			}
```

Comment:
This helps filter out the extra `setNext:` calls that the client is doing.
I've done the change in here because it helps improve the caching logic (repeated setNext calls to the same were causing some issues)

Reply:
How about adding this one behind a dev FF or actual FF? (Not sure there could be some scenario we are missing to make it risky. Maybe the only one that needs testing is when we repeat 🤔)

Reply:
I can put it behind the same FF as the new caching behavior so at least they are tied and we can revert to the current state. 
We have lots of flags active that we need to start confirming that everything is ok and removing 😅